pkg/apiserver/proxy.go

```diff
@@ -286,6 +286,11 @@ func dialURL(url *url.URL, transport http.RoundTripper) (net.Conn, error) {
 			return nil, err
 		}
 
+		// Return if we were configured to skip validation
```

Comment:
Since this wasn't strictly necessary for this PR, I split it out to https://github.com/kubernetes/kubernetes/pull/14889 so it can be reviewed with it's tests there.